summary-v4-object.go

```diff
@@ -0,0 +1,166 @@
+//
+// Copyright (c) 2015-2025 MinIO, Inc.
+//
+// This file is part of MinIO Object Storage stack
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+//
+
+package madmin
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+// ObjectSummaryOptions ...
+type ObjectSummaryOptions struct {
+	Bucket, Object string
+}
+
+// ObjectSummary calls minio to search for all files and parts
+// related to the given object, across all disks.
+func (adm *AdminClient) ObjectSummary(ctx context.Context, objOpts ObjectSummaryOptions) (objectSummaryResponse *ObjectSummaryRespose, err error) {
+	form := make(url.Values)
+	if objOpts.Bucket != "" {
+		form.Add("bucket", objOpts.Bucket)
+	}
+	if objOpts.Object != "" {
+		form.Add("object", objOpts.Object)
+	}
```

Comment:
what is the behavior when bucket is not provided? or object is not provided?

Reply:
I can make some kind of "error" here so it won't call minio.. right now it just calls minio and minio throws an error